### DISCUSSION TITLE
Any plan to build Positron on AI-powered IDE

### DISCUSSION THREAD

**Discussion by ntluong95** [posit-dev/positron#5549] [Feedback] [answered]
Recently I have known about Windsurf AI and Cursor IDE, both are a fork of CodeOSS project like Positron, but they support a lot of AI-powered model. Personally I don't like them as they are not a data science oriented IDE.

While depending on Github Copilot seems futile, should Positron develop their own AI feature supporting for data science tasks. In the recent release of VSCode 1.95, Microsoft has introduced a bunch of extensions, making work with data is much easier

**Comment by juliasilge** [comment, marked as answer]
EDIT by TM on 2025-06-03: Please see https://positron.posit.co/assistant.html for information on Positron Assistant, a native GenAI client for Positron.

---

We are definitely interested in integrating LLM coding assistant support in Positron, even perhaps in a more integrated way than possible just with the VS Code extension API. We will be sure to share details in release notes, etc, as this work gets underway!

**Reply by CorneeldH** [reply]
Very big upvote. A future IDE should not just be AI-assisted, but AI-native. As in: you talk in the IDE and get code back and repeat until satisfied. This has impact on the UI as well, it means for instance that buttons are no longer needed.

**Reply by ethanmed** [reply]
*Great!* I am very much looking forward to the upcoming big steps in Positron! I am a huge fan of R from rural China

**Reply by henrydehe** [reply]
A mix of Positron and Cursor would be an absolute game-changer.

**Comment by Guiquan-27** [comment]
I am very much looking forward to having such technological enhancements! To be honest, I really like IDEs similar to Cursor. However, they are not user-friendly for data scientists, particularly in terms of data viewing, plot presentation, and some specific requirements of the R language.

**Comment by jiangshiguo** [comment]
I recently switched to cursor pro. With the AI assistant, the programming experience is way much better than RStudio, Positron, and VS Code. Cursor and VS Code have decent support for data inspection as well, even though the support for Python is better with the Data Wrangler extension.

**Comment by matthewgson** [comment]
Just a question - are there any significant differences between using those IDEs and using extensions in Positron (e.g., Codium extension).

**Comment by paulovillarroel** [comment]
It would definitely be a game changer if Positron is integrated to some LLM (like Cursor or Windsurf) as an agent, but focused on data science. That's something that's not on the market and would be great!!!

**Reply by MartinSclxmedia** [reply]
Yes, I would appreciate MCP style support in Positron

**Comment by kendavidn** [comment]
Just here to cheer you guys on. Cursor is the fastest-growing SaaS of all time, supposedly. 

A similar setup for data science folks would be huge unlock and Positron is best placed to take this on.

**Reply by ntluong95** [reply]
Hi Kene. I believe they are coming up with Positron Assistant, should be available soon.

**Comment by jhelvy** [comment]
A basic Cursor subscription is $20/month. It's a quite impressive coding experience, but nowhere near as good an experience for data work. Positron is a strictly better IDE, but the AI-assisted coding in Cursor is top notch. I expect many would pay for a pro Positron subscription that blends these - a Positron version that has more seamlessly integrated AI assistance like Cursor. It could become a significant revenue stream.

**Reply by MartinSclxmedia** [reply]
I agree with your assessment. We pay $40 per seat/month for cursor and I would prefer doing this work in Positron.

**Comment by davidrsch** [comment]
Hello! From what I’ve seen in the issues, it looks like the Positron AI Assistant is being actively developed, really exciting stuff! However, I haven’t been able to find any documentation or guide on how to add or configure custom models for testing. Is there a way to try this out already, or is that feature not yet available to the public? Thanks in advance for any pointers!

**Reply by jmcphers** [reply]
Yes, we're working on it! It isn't available to the public yet. We'll post here when we have something you can try out!

**Reply by davidkane9** [reply]
Is there any way for outsiders to follow the development of Positron AI Assistant, like via the NEWS file in a GitHub repo somewhere? I am especially interested in a timeline for alpha testing since my summer course starts in two months.

**Reply by juliasilge** [reply]
I don't think we would recommend using Positron in a teaching environment until we are out of beta, which is currently scheduled for mid summer sometime.

If you are interested in keeping up with what we are up to in terms of features and such, we do publish some release notes with each GH release here: https://github.com/posit-dev/positron/releases

**Reply by davidkane9** [reply]
Thanks. That is very helpful.

**Comment by lawremi** [comment]
This probably changes with co-pilot being open-sourced.

**Comment by davidrsch** [comment]
Hello I will like to share that [Gemini code assistant](https://open-vsx.org/extension/Google/geminicodeassist) is available already, it will be great if it could be integrated with copilot or Positron assitant when realesed instead of having a separate extension for it. Bu it offers a reliable alternative while waiting for Positron assistant, which for I can see in the issues and pull request will arrive with extra features like taking concept from variables in the environment.
